kafka/src/test/java/io/specmesh/kafka/ExporterFunctionalTest.java

```diff
@@ -0,0 +1,159 @@
+                            Domain.LIMITED.domainId + "-secret")
+                    .withKafkaAcls(aclsForOtherDomain(Domain.LIMITED))
+                    .withKafkaAcls(aclsForOtherDomain(Domain.UNRELATED))
+                    .build();
+
+    @BeforeAll
+    static void setUp() {
+        try (Admin adminClient = KAFKA_ENV.adminClient()) {
+            Provisioner.provisionTopics(API_SPEC, adminClient);
+            Provisioner.provisionAcls(API_SPEC, adminClient);
+        }
+    }
+
+    @Test
+    void shouldExportAPIFromCluster() throws Exporter.ExporterException {
+
+        try (Admin adminClient = KAFKA_ENV.adminClient()) {
+            final ApiSpec exported = Exporter.export(aggregateId, adminClient);
+            assertThat(
+                    exported.channels().keySet(),
+                    containsInAnyOrder(
+                            "london.hammersmith.olympia.bigdatalondon._protected.retail.subway.food.purchase",
+                            "london.hammersmith.olympia.bigdatalondon._public.attendee",
+                            "london.hammersmith.olympia.bigdatalondon._private.retail.subway.customers"));
+        }
+    }
+
+    @Test
+    void shouldExportYAMLAPIFromCluster() throws Exporter.ExporterException {
+
+        final ApiSpec apiSpec =
+                ApiSpec.builder()
+                        .id("urn:asyncapi-id")
+                        .version("version-123")
+                        .channels(
+                                Map.of(
+                                        "one-topic-channel",
+                                        Channel.builder()
+                                                .description("one-topic-channel-description")
+                                                .bindings(
+                                                        Bindings.builder()
+                                                                .kafka(
+                                                                        KafkaBinding.builder()
+                                                                                .groupId(
+                                                                                        "kafka-binding-group-id")
+                                                                                .build())
+                                                                .build())
+                                                .build()))
+                        .build();
+        final String specYaml = Exporter.exportYaml(apiSpec);
+        assertThat(specYaml, is(containsString("id: \"urn:asyncapi-id\"")));
+        assertThat(specYaml, is(containsString("version: \"version-123\"")));
+        assertThat(specYaml, is(containsString("one-topic-channel:")));
+
+        assertThat(specYaml, is(containsString("description: \"one-topic-channel-description\"")));
+        assertThat(specYaml, is(containsString("groupId: \"kafka-binding-group-id\"")));
```

Comment:
Still pretty weak testing.

I'd suggest having the expected spec in a yaml file under test resources. Load this file here and compare to the actual generated spec, failing if different.

That way, you _know_ the spec is what you want.

Also, consider having another test to _parses_ the output again, to ensure its valid YAML.

Reply:
ok - will do that